### DISCUSSION TITLE
Add blur node to be built-in

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#10592] [2D]
Well, this feature is basically going to add a new built-in node to the engine, called "Blur2D". This is similar to Unreal Engine's UI Blur widget, where the blur effect is added seamlessly, and there are common parameters to adjust, e.g. blur type, strength, radius,...
**NOTE**: _I know having multiple blur types all in one node will probably cause some performance issues, so to resolve this, how about we add a "Blur2D" class/category inside "2D", then inside that place various types of blur implementations. This should help readibility too._

This Blur node is going to add the effect at the **2D Screen Level** to GUI nodes(Button's, Label's,...), and 2D nodes(TileMapLayer's, CharacterBody2D, ...). In a 3D scene, the blur effect will apply to the whole camera/viewport.

The node is inspired from Unreal Engine's UI widget.

_Image from Unreal Engine showcasing the blur widget, and some of its parameters_
![ue_blur_example](https://github.com/user-attachments/assets/8b81e0d3-00db-42c9-a684-dd172bca662b)

I know this can be achieved through shaders... But still, I think alot of game-devs/designers would really enjoy having a simple way(node) to add blur effect, and change some easy parameters to get the desired effect without getting their hands dirty deep in the logic/implementation behind the effect itself.

I'm open to the idea of Blur3D, but, the fact is it's going to be alot more complex than 2D, and it could get messy fast, with the conflicts of different desired designs/results. So, I say we stick to just implementing Blur2D.

I don't have a working prototype, nor am I knowledgeable with Godot's code-base..., but I'm willing to test and give feedback to any implementations for this feature.

I'm willing to hear your thoughts, and how to improve the idea!

**Comment by Mickeon** [comment]
You may be looking for this addon: https://github.com/sphynx-owner/JFA_driven_motion_blur_addon

- See also https://github.com/godotengine/godot/pull/92711

**Reply by Meorge** [reply]
Perhaps I'm misunderstanding the post, but I was thinking they were talking less about motion blur and more about achieving a blurred UI effect like this (image from https://forum.juce.com/t/faster-blur-glassmorphism-ui/43086):
![image](https://github.com/user-attachments/assets/c41f3e2a-fa9c-4461-b03a-024c2dc552e9)

**Reply by ghost** [reply]
Yes, I was just going to say this, the image is good representation, I know motion blur is important, but it's not what im talking about.

Another example relating to Blur2D, e.g. In overwatch main menu there is this pic with no blur:
![overwatch-4](https://github.com/user-attachments/assets/e3d1486c-2a5f-4776-aaf1-0d665c1505d5)

After you click the play button there is a nice transition to the menu, the menu has a background blur, I wouldn't call it MotionBlur, just a simple Blur2D node:
![resize](https://github.com/user-attachments/assets/251d35ce-2d0e-4309-b7b0-f85ca26fb6f2)

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/3485.

**Reply by ghost** [reply]
I don't think the issue you mentioned is the same refered in this discussion.

#3485 wants to add another property(blur parameters) to the StyleBoxFlat resource, as mentioned
"_Add a "Blur" option and dropdown menu to the StyleBoxFlat._" - @likeich

I believe his idea as much as it may be ok, still it would cause a few problems, one of which I may say is this would cause the resource to be used as a place to have all different effects placed in it, not just blur. It would make it more complex and convoluted.

In this discussion I'm hoping for a new separate node/group of nodes that specificly give the desired effect of blur with all of its parameters/options, thus the dev can easily manage the effect.

 Also, I agree with @likeich, blur has become so common and important, there should be a fairly easy way for designers in godot to add it without having to deal with coding it themself or resorting to an addon, hence its a commonly needed effect.
